### DISCUSSION TITLE
adc_dodule error [solved]

### DISCUSSION THREAD

**Discussion by Flourdeau** [marcel-licence/esp32_basic_synth#17] [General]
hi, since i have downloaded the new repo, i have this error:

Arduino : 1.8.13 (Windows Store 1.8.42.0) (Windows 10), Carte : "ESP32 Dev Module, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 921600, None"

C:\Users\fred\Downloads\esp32_basic_synth-main (1)\esp32_basic_synth-main\esp32_basic_synth\adc_module.ino: In function 'void AdcMul_Init()':

adc_module:56:22: error: 'analogSetCycles' was not declared in this scope

     analogSetCycles(1);

                      ^

C:\Users\fred\Downloads\esp32_basic_synth-main (1)\esp32_basic_synth-main\esp32_basic_synth\adc_module.ino: In function 'void AdcMul_Process()':

adc_module:96:20: error: 'adcStart' was not declared in this scope

         adcStart(12);

                    ^

adc_module:100:27: error: 'adcBusy' was not declared in this scope

             if (adcBusy(12) == false)

                           ^

adc_module:102:38: error: 'adcEnd' was not declared in this scope

                 readAccu += adcEnd(12);

                                      ^

adc_module:106:18: error: 'adcEnd' was not declared in this scope

         adcEnd(12);

                  ^

Plusieurs bibliothèque trouvées pour "WiFi.h"

Utilisé : C:\Users\fred\Documents\ArduinoData\packages\esp32\hardware\esp32\1.0.6\libraries\WiFi

Non utilisé : C:\Program Files\WindowsApps\ArduinoLLC.ArduinoIDE_1.8.42.0_x86__mdqgnx93n4wtt\libraries\WiFi

exit status 1

'analogSetCycles' was not declared in this scope

**Comment by marcel-licence** [comment]
Salut,

I guess the required libraries are missing. 
I could extract following from my build log:

Resolved libraries:
  WiFi.h [WiFi@1.2.7 WiFi@1.0]
  AC101.h [AC101@0.0.1]
  Wire.h [Wire@1.0.1]

**Reply by marcel-licence** [reply]
I am using an older version of the esp library:
esp32\1.0.4

They removed the functions from the latest version.

**Reply by pc0rba** [reply]
1.0.4 
not available for installation.
not listed
3..5

**Reply by marcel-licence** [reply]
![grafik](https://user-images.githubusercontent.com/65245426/119254258-9fd5dd80-bbb5-11eb-89bb-742e09e8a476.png)

It can be found in the board manager. I will update soon to 1.0.6 and try to make adc working with the latest version of theboard libraries.

**Reply by Flourdeau** [reply]
Thx marcel. 
No compilation errors, problem solved for me.😎

**Reply by marcel-licence** [reply]
Good to hear =)
